codec-http2/src/test/java/io/netty/handler/codec/http2/DataCompressionHttp2Test.java

```diff
@@ -110,7 +112,18 @@ public void teardown() throws InterruptedException {
 
     @Test
     public void justHeadersNoData() throws Exception {
-        bootstrapEnv(0);
+        // The stream will be closed upon creation. We initialize the server latch to 2 because
+        // we ensure that 1. the stream is closed, and 2. that we received a headers read.
+        bootstrapEnv(0, 2);
+        doAnswer(new Answer<Void>() {
+            @Override
+            public Void answer(InvocationOnMock invocation) throws Throwable {
+                // Trigger point 2 to decrement serverLatch, and so the test doesn't prematurely exit.
+                serverLatch.countDown();
```

Comment:
@nmittler - The issue was that because this stream is created in the half-closed state (headers frame indicates EOS) that `onStreamActive` is called when `stream.state()` is half closed, and thus `onStreamHalfClosed` does not get called (because no state change).

The expected sequence of events is:

1. onStreamActive
2. onHeadersRead

So if we allow the `onStreamActive` to indicate we are done with the test, we are in a race condition to check the `verify(serverListener).onHeadersRead(...)` condition.

This is admittedly hacky, and I recall trying to simplify the trigger points in the past by adding additional methods to `Http2Connection` and also calling `onStreamHalfClosed` (even with no stream transition) but I think we decided to instead add a comment for the `onStreamHalfClosed` method:

```java
         * This method will <strong>not</strong> be called until a state transition occurs from when
         * {@link #onStreamActive(Http2Stream)} was called.
```